daemon-scheduler/pkg/api/v1/api_test.go

```diff
@@ -64,13 +70,119 @@ func (suite *APITestSuite) TestPing() {
 	assert.Equal(suite.T(), http.StatusOK, responseRecorder.Code, "Http response status in ping response is invalid")
 }
 
-func (suite *APITestSuite) getRouter() *mux.Router {
-	r := mux.NewRouter().StrictSlash(true)
-	s := r.Path("/v1").Subrouter()
+func (suite *APITestSuite) TestGetEnvironmentReturnsError() {
+	name := "testEnv"
+	err := errors.New("Error from GetEnvironment")
+	suite.environment.EXPECT().GetEnvironment(gomock.Any(), name).Return(nil, err)
+	request, err := http.NewRequest("GET", "/v1/environments/"+name, nil)
```

Comment:
I'd move this to a separate function since it's being used in multiple functions and any changes to it later will not be hard -
func (suite *APITestSuite) generateGetEnvRequest(t *testing.T, name string) *http.Request {
    request, err := http.NewRequest("GET", "/v1/environments/"+name, nil)
    assert.Nil(suite.T(), err, "Unexpected error")
    return request
}